src/traceViewer/web/ui/sourceTab.tsx

```diff
@@ -38,65 +40,36 @@ export const SourceTab: React.FunctionComponent<{
     return { fileName, lineNumber };
   }, [actionEntry]);
 
-  const content = useAsyncMemo<string>(async () => {
-    if (location.fileName)
-      return window.readFile(location.fileName);
-    return location.value || '';
-  }, [location.fileName, location.value], '');
-
-  const [editor, setEditor] = React.useState<{
-    editor: MonacoEditor.IStandaloneCodeEditor,
-    element: HTMLElement,
-    decorations: string[]
-  } | undefined>();
-  const [measure, ref] = useMeasure<HTMLDivElement>();
-
-  React.useLayoutEffect(() => {
-    if (!ref.current)
-      return;
-    if (editor && editor.element === ref.current)
-      return;
-
-    const standalone = monaco.editor.create(ref.current, {
-      value: '',
-      language: 'javascript',
-      readOnly: true
-    });
-    standalone.layout();
-    setEditor({ editor: standalone, element: ref.current, decorations: [] });
-  }, [ref, editor]);
-
-  React.useLayoutEffect(() => {
-    if (editor) {
-      editor.editor.setValue(content);
-      editor.decorations = decorateLine(editor.editor, location.lineNumber, editor.decorations);
+  const content = useAsyncMemo<string[]>(async () => {
+    const value = location.fileName ? await window.readFile(location.fileName) : location.value
+    const result = [];
+    let continuation: any;
+    for (const line of (value || '').split('\n')) {
```

Comment:
How fast is highlighting? I'd feel safer if there was a length check here and it bails out instead of highlighting 10mb on the main thread.

Reply:
We'll worry about it at some point!